lib/SetAPI/generic/SetInterfaceV1.py

```diff
@@ -103,43 +215,100 @@ def get_set(
 
         :param self: this class
         :type self: SetInterfaceV1
-        :param ref: object reference
-        :type ref: str
-        :param include_item_info: whether or not to populate the info of the set items, defaults to False
-        :type include_item_info: bool, optional
-        :param ref_path_to_set: ref path for the object, defaults to None
-        :type ref_path_to_set: list[str] | None, optional
-        :param include_set_item_ref_paths: whether or not to add ref paths, defaults to False
-        :type include_set_item_ref_paths: bool, optional
+        :param params: dictionary of parameters for the request
+        :type params: dict[str, Any]
         :return: set data
         :rtype: dict[str, Any]
         """
-        if ref_path_to_set is None:
-            ref_path_to_set = []
-        obj_selector = build_ws_obj_selector(ref, ref_path_to_set)
+        checked_params = self._check_get_set_params(params)
 
-        # retrieve the object from the workspace
-        ws_data = self.ws.get_objects2({"objects": [obj_selector]})
+        obj_selector = build_ws_obj_selector(
+            checked_params["ref"], checked_params[REF_PATH_TO_SET]
+        )
 
+        resp = self.ws_client.get_objects2({"objects": [obj_selector]})
         set_data = {
-            "data": ws_data["data"][0]["data"],
-            "info": ws_data["data"][0]["info"],
+            "data": resp["data"][0]["data"],
+            "info": resp["data"][0]["info"],
         }
+        kbase_set_type = set_data["info"][2]
+
+        if "KBaseSets" not in kbase_set_type:
+            return self._populate_non_kbase_set(
+                obj_selector, kbase_set_type, set_data, checked_params
+            )
 
         # not all sets have 'items' (e.g. KBaseSearch.GenomeSet);
         # if they don't, skip the following steps as they will error out
         if not set_data["data"].get("items", []):
             return set_data
 
-        if include_item_info:
-            self._populate_item_object_info(set_data, ref_path_to_set)
+        if checked_params[INC_ITEM_INFO]:
+            self._populate_item_object_info(set_data, checked_params[REF_PATH_TO_SET])
 
-        if include_set_item_ref_paths:
+        if checked_params[INC_ITEM_REF_PATHS]:
             set_items = set_data["data"]["items"]
             populate_item_object_ref_paths(set_items, obj_selector)
 
         return set_data
 
+    def _check_get_set_params(
+        self: "SetInterfaceV1", params: dict[str, Any]
+    ) -> dict[str, Any]:
```

Comment:
this code is identical to the `_check_get_set_params` that all the `get_***_set` functions executed, back in the days when there were classes for each set type